include/bluetooth/audio/vcs.h

```diff
@@ -0,0 +1,504 @@
+ */
+int bt_vcs_get(struct bt_conn *conn, struct bt_vcs *service);
+
+/**
+ * @brief Callback function for bt_vcs_discover
+ *
+ * @param conn         The connection that was used to discover
+ *                     Volume Control Service.
+ * @param err          Error value. 0 on success, GATT error on positive value
+ *                     or errno on negative value.
+ * @param vocs_count   Number of Volume Offset Control Service instances
+ *                     on peer device.
+ * @param aics_count   Number of Audio Input Control Service instances on
+ *                     peer device.
+ */
+typedef void (*bt_vcs_discover_cb)(struct bt_conn *conn, int err,
+				   uint8_t vocs_count, uint8_t aics_count);
+
+/**
+ * @brief Callback function for Volume Control Service volume state.
+ *
+ * Called when the value is read,
+ * or if the value is changed by either the server or client.
+ *
+ * @param conn    The connection to the peer device.
+ * @param err     Error value. 0 on success, GATT error on positive value
+ *                or errno on negative value.
+ * @param volume  The volume of the Volume Control Service server.
+ * @param mute    The mute setting of the Volume Control Service server.
+ */
+typedef void (*bt_vcs_state_cb)(struct bt_conn *conn, int err, uint8_t volume,
+				uint8_t mute);
+
+/**
+ * @brief Callback function for Volume Control Service flags.
+ *
+ * Called when the value is read,
+ * or if the value is changed by either the server or client.
+ *
+ * @param conn    The connection to the peer device.
+ * @param err     Error value. 0 on success, GATT error on positive value
+ *               or errno on negative value.
+ * @param flags   The flags of the Volume Control Service server.
+ */
+typedef void (*bt_vcs_flags_cb)(struct bt_conn *conn, int err, uint8_t flags);
+
+/**
+ * @brief Callback function for writes.
+ *
+ * @param conn    The connection to the peer device.
+ * @param err     Error value. 0 on success, GATT error on fail.
+ */
+typedef void (*bt_vcs_write_cb)(struct bt_conn *conn, int err);
+
+struct bt_vcs_cb {
+	/* Volume Control Service */
+	bt_vcs_state_cb               state;
+	bt_vcs_flags_cb               flags;
+#if defined(CONFIG_BT_VCS_CLIENT)
+	bt_vcs_discover_cb            discover;
+	bt_vcs_write_cb               vol_down;
+	bt_vcs_write_cb               vol_up;
+	bt_vcs_write_cb               mute;
+	bt_vcs_write_cb               unmute;
+	bt_vcs_write_cb               vol_down_unmute;
+	bt_vcs_write_cb               vol_up_unmute;
+	bt_vcs_write_cb               vol_set;
+
+	/* Volume Offset Control Service */
+	struct bt_vocs_cb               vocs_cb;
+
+	/* Audio Input Control Service */
+	struct bt_aics_cb               aics_cb;
+#endif /* CONFIG_BT_VCS_CLIENT */
+};
+
+/**
+ * @brief Discover Volume Control Service and included services.
+ *
+ * This will start a GATT discovery and setup handles and subscriptions.
+ * This shall be called once before any other actions can
```

Comment:
Insert "be".
Before other actions can be completed, or before other actions can be started?

Reply:
changed complete to `executed`